### ISSUE TITLE
Replied Persistent Notification Not Cleared

### ISSUE THREAD

**Issue by amosyuen** [home-assistant/android#3150]
**Home Assistant Android version:** 2022.11.0-full

**Android version:** 13

**Phone model:** Pixel 6

**Home Assistant version:** 2022.12.4

**Last working Home Assistant release (if known):**

**Description of problem:** Can not clear a persistent notification that has been replied. After replying there is a spinning loading symbol seemingly indefinitely. Then When I call clear_notification that symbol disappears and the reply shows inline. Calling clear_notification a second time does not do anything.

Call to create notification:
```yaml
service: notify.mobile_app_amos_phone
data:
  message: Alarm
  data:
    tag: alarm
    persistent: true
    sticky: true
    actions:
      - action: REPLY
        title: Disarm with Pin
```

Call to clear notification:
```yaml
service: notify.mobile_app_amos_phone
data:
  message: clear_notification
  data:
    tag: alarm
```

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```
12-13 23:10:31.934 32095 32111 E BLASTBufferQueue: [VRI[SettingsActivity]#2](f:0,a:3) Faking releaseBufferCallback from transactionCompleteCallback
12-13 23:10:36.687 32095 32420 W FirebaseMessaging: Unable to log event: analytics library is missing
12-13 23:10:36.687 32095 32420 D FCMService: From: 331041709873
12-13 23:10:36.689 32095 32095 D MessagingService: Creating notification with following data: {sticky=true, tag=alarm, persistent=true, message=Alarm, action_1_title=Disarm with Pin, action_1_key=REPLY}
12-13 23:10:36.693 32095 32095 D MessagingService: Show notification with tag "alarm" and id "92895825"

...

12-13 23:10:51.274 32095 32129 E BLASTBufferQueue: [VRI[WebViewActivity]#3](f:0,a:4) Faking releaseBufferCallback from transactionCompleteCallback
12-13 23:10:51.280 32095 32125 D NotifManagerCompat: Cancel notification with tag "null" and id "-1"
12-13 23:10:51.281 32095 32125 D NotifManagerCompat: Check if the notification is in a group...
12-13 23:10:51.281 32095 32125 D NotifManagerCompat: Notification is not in a group. Cancel notification...
12-13 23:10:51.292 32095 32129 E BLASTBufferQueue: [VRI[WebViewActivity]#3](f:0,a:4) Faking releaseBufferCallback from transactionCompleteCallback
```

**Screenshot of problem:**

After submitting reply
![image](https://user-images.githubusercontent.com/422794/207530012-6dc97e34-b13e-41b1-93fb-80c031ed34ec.png)

After sending clear_notification
![image](https://user-images.githubusercontent.com/422794/207530034-971e2602-edd9-4991-8a5a-1053c93492ee.png)

**Additional information:**

If I have not replied to the notification, clearing works correctly.

Also if I send a new notification with the same tag it is replaced correctly.

**Comment by dshokouhi** [comment]
I was able to reproduce part of this issue, my replies are sent successfully and the wheel does not spin indefinitely however the notification remains.

Judging from the code and stack overflow responses this is working as designed for Android and in order to clear the notification you need to first update the notification.

What appears to be happening is that the notification is cleared out of the system but Android itself will hold on to the notification because it was replied to. This issue also occurs even when persistent is not used, its related to the reply and Android keeping the notification. So the code does indeed clear the notification as we can see even android does not consider it to be an active notification, its only visible when you pull down the panel if you notice the icon no longer shows when the panel is not open after the reply.

https://stackoverflow.com/questions/51736472/android-p-notificationmanager-cancel-does-not-work-if-user-has-replied-to-not
https://stackoverflow.com/questions/54219914/cancel-notification-with-remoteinput-not-working

**Comment by jpelgrom** [comment]
While this may be correct from Android's point of view, as a user you'd expect to be able to cancel the notification.

[The documentation mentions that the notification should be updated after handling the direct reply](https://developer.android.com/develop/ui/views/notifications/build-notification#:~:text=After%20you%E2%80%99ve%20processed%20the%20text%2C%20you%20must%20update%20the%20notification%20by%20calling%20NotificationManagerCompat.notify()%20with%20the%20same%20ID%20and%20tag%20(if%20used).), which the app currently doesn't do. If it did it is probably possible to cancel it again.

> After you’ve processed the text, you must update the notification by calling `NotificationManagerCompat.notify()` with the same ID and tag (if used).

**Comment by dshokouhi** [comment]
I saw that but on a Pixel the UI is already hidden and confirmed to the user that it sent with its own update.

> This is necessary to hide direct reply UI and confirm to the user that their reply was received and processed correctly.

I believe what they mean by update the notification is to add the response to the text just like a messaging app does, because updating the notification will remove the response that was added if we don't change it. Which will appear as if the reply was sent but nothing happened. This allows chat apps to keep showing new replied messages, feels like the purpose is a bit different for us here.

**Comment by jpelgrom** [comment]
I managed to change the code so that it updates the notification on reply instead of asking the system to remove it (which it doesn't), which allows you to update/cancel it after entering a response like expected.

The only regression I've found is that it will now only show the latest reply, instead of x replies (the system decides how many to show, 1 to 3 in my testing). Would that create problems or be considered a breaking change?
